enhancements/operator-hub-check.md

```diff
@@ -0,0 +1,103 @@
+| To check if the operator bundle is using the Webhook API version which is deprecated | warning | Check if the CSV has the spec `webhookdefinitions.admissionReviewVersions.v1beta1`. (See [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml#L203-L205). Note that in this cases users should be informed that the `admissionregistration.k8s.io/v1beta1` was deprecated in Kubernetes `1.16` and will be removed in `1.22`. 
+| To check if the operator bundle is using the CRD API version which is no longer supported | error | If `spec.minKubeVersion` is informed and it is >= `1.22` then, check if the CRD manifests in the bundle are using `apiVersion: apiextensions.k8s.io/v1beta1`. (See [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/cache.example.com_memcacheds.yaml#L1) this manifest in the SDK testdata go sample using `go/v2`). Note that in this cases users should be informed that the `apiextensions/v1beta1` was deprecated in Kubernetes `1.16` and was removed in `1.22`.
+| To check if the operator bundle is using the Webhook API version which is no longer supported | error | If `spec.minKubeVersion` is  informed  and it is >= `1.22` then, check if the CSV has the spec `webhookdefinitions.admissionReviewVersions.v1beta1`. (See [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml#L203-L205). Note that in this cases users should be informed that the `admissionregistration.k8s.io/v1beta1` was deprecated in Kubernetes `1.16` and it is removed in `1.22`. 
+
+**Recommendations and conventions**
+
+| Check | Type | Description | 
+| ------ | ----- | ----- |
+| To check if the operator bundle versioning convention is followed | warning | Check if the operator bundle name (`metadata.name`) in CSV matches with `operatorname-v0.0.0`. We can use regex and semver go lib to do this check. 
+| To check if the operator bundle channel names are following up the definition. See [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/channel-naming.md#recommended-channel-naming) | warning | Check if the channels name are not in `preview`,`stable`,`fast` or in `preview-*`,`stable-*`,`fast-*`. We can use regex and raise the warning with the link for the doc. Also the warning should return all values that does not matches with criteria. See that the channel names can be found in the index image (e.g [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/metadata/annotations.yaml#L2)) or in the package (e.g [here](https://github.com/operator-framework/community-operators/blob/master/upstream-community-operators/mysql/mysql.package.yaml#L1))
```

Comment:
i'm not sure if we want to get to the point of validating channel names (beyond syntactic validity).  While we have our best practices, i'm leery of trying to enforce them in a validator.

Reply:
My idea here is check who is following the convention and then, raise a WARNING only if it is not followed. So, users will still able to use any name as please them. However, we will have an opportunity to notify them about the convention and do the recommendations automatically.

Reply:
It is one of the requirements. So, I am closing this as solved. Let me know if you still with any objection.

Reply:
It is a requirement of what?